### PR TITLE
feat: Flux added functionaities

### PR DESCRIPTION
**Pull request by RajeevRanjan27**
# Description

Fixes https://github.com/devtron-labs/devtron/issues/5332

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: api/k8s/application/k8sApplicationRestHandler.go [hunk 1]
```diff
@@ -215,6 +217,33 @@ func (handler *K8sApplicationRestHandlerImpl) GetResource(w http.ResponseWriter,
...
+		if err != nil || !valid {
+			handler.logger.Errorw("error in validating resource request", "err", err)
+			common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+			return
+		}
+
+		//RBAC enforcer starts here
+
+		if ok := handler.enforcer.Enforce(token, casbin.ResourceGlobal, casbin.ActionGet, "*"); !ok {
+			common.WriteJsonResp(w, errors.New("unauthorized"), nil, http.StatusForbidden)
```
[the remark this answers is not in the document]
Comment (prakash100198) [reply api/k8s/application/k8sApplicationRestHandler.go:242]: Do everywhere as written above, extract out in a rbac func, 
in most of the cases one rbac func will work for all

**Review by prakash100198 — CHANGES_REQUESTED** [review]

File: api/k8s/application/k8sApplicationRestHandler.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (RajeevRanjan27) [reply api/k8s/application/k8sApplicationRestHandler.go]: yes, extracted and done for the get, update and delete related queries, as it invloves the validation of resources as well

File: api/k8s/application/k8sApplicationRestHandler.go [hunk 2]
```diff
@@ -162,59 +165,19 @@ func (handler *K8sApplicationRestHandlerImpl) GetResource(w http.ResponseWriter,
...
-		if request.DeploymentType == bean2.HelmInstalledType {
-			//TODO Implement ResourceRequest Validation for Helm Installed Devtron APPs
-		} else if request.DeploymentType == bean2.ArgoInstalledType {
-			//TODO Implement ResourceRequest Validation for ArgoCD Installed APPs From ResourceTree
-		}
-		// RBAC enforcer applying for Devtron App
-		envObject = handler.enforcerUtil.GetEnvRBACNameByAppId(request.DevtronAppIdentifier.AppId, request.DevtronAppIdentifier.EnvId)
-		hasReadAccessForEnv := handler.enforcer.Enforce(token, casbin.ResourceEnvironment, casbin.ActionGet, envObject)
-		if !hasReadAccessForEnv {
+		} else if ok == false {
```
Comment (prakash100198) [inline api/k8s/application/k8sApplicationRestHandler.go:177]: else if ok == false -> else if !ok

File: api/k8s/application/k8sApplicationRestHandler.go [hunk 3]
```diff
@@ -266,52 +229,94 @@ func (handler *K8sApplicationRestHandlerImpl) GetResource(w http.ResponseWriter,
...
 	common.WriteJsonResp(w, nil, resource, http.StatusOK)
 }
-
 func (handler *K8sApplicationRestHandlerImpl) GetHostUrlsByBatch(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	clusterIdString := vars["appId"]
 	if clusterIdString == "" {
 		common.WriteJsonResp(w, fmt.Errorf("empty appid in request"), nil, http.StatusBadRequest)
 		return
 	}
```
Comment (prakash100198) [inline api/k8s/application/k8sApplicationRestHandler.go:238]: appIdString

File: pkg/fluxApplication/FluxApplicationService.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	req.ClusterConfig = conf
+	res, err := impl.helmAppClient.Hibernate(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	response := service.HibernateResponseAdaptor(res.Status)
+	return response, nil
+}
+
+func (impl *FluxApplicationServiceImpl) UnHibernateFluxApplication(ctx context.Context, app *bean.FluxAppIdentifier, hibernateRequest *openapi.HibernateRequest) ([]*openapi.HibernateStatus, error) {
```
Comment (prakash100198) [inline pkg/fluxApplication/FluxApplicationService.go:62]: log error using impl.logger.Errorw

**Review by prakash100198 — CHANGES_REQUESTED** [review]

**Comment by RajeevRanjan27** [comment]
https://github.com/devtron-labs/devtron/pull/5528

**Pull request closed without merging**